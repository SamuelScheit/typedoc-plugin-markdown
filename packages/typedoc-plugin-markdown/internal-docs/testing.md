---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
