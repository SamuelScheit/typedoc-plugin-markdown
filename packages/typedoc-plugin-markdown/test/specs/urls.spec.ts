import { expectUrlsToEqual } from '@devtools/testing';
describe(`Urls`, () => {
  test(`should gets Urls for multiple entry points`, () => {
    expectUrlsToEqual('groups', ['members', 'modules']);
  });

  test(`should gets Urls for single entry points`, () => {
    expectUrlsToEqual('reflections', ['members', 'modules']);
  });

  test(`should gets Urls for packages entry points`, () => {
    expectUrlsToEqual('packages', ['members', 'modules']);
  });

  test(`should gets Urls for single package entry points`, () => {
    expectUrlsToEqual('package', ['members', 'modules']);
  });

  test(`should gets Urls for entry module`, () => {
    expectUrlsToEqual('entryfiles', ['members', 'modules']);
  });

  test(`should gets Urls for readme options`, () => {
    expectUrlsToEqual('readme', ['members', 'modules']);
  });

  test(`should gets Urls for modules parts`, () => {
    expectUrlsToEqual('modules', ['members', 'modules']);
  });
});
