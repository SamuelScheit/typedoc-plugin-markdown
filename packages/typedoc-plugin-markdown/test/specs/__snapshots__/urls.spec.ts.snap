// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Urls should gets Urls for entry module: outputFileStrategy: members 1`] = `
[
  "README.md",
  "entry-module/classes/ModuleClass.md",
  "module-1/README.md",
  "module-1/classes/ModuleClass.md",
  "module-2/README.md",
  "module-2/classes/ModuleClass.md",
  "modules.md",
  "readme_.md",
]
`;

exports[`Urls should gets Urls for entry module: outputFileStrategy: members 2`] = `
[
  "entry-module/classes/ModuleClass.md",
  "index.md",
  "module-1/classes/ModuleClass.md",
  "module-1/index.md",
  "module-2/classes/ModuleClass.md",
  "module-2/index.md",
  "modules.md",
]
`;

exports[`Urls should gets Urls for entry module: outputFileStrategy: modules 1`] = `
[
  "README.md",
  "module-1.md",
  "module-2.md",
  "modules.md",
  "readme_.md",
]
`;

exports[`Urls should gets Urls for entry module: outputFileStrategy: modules 2`] = `
[
  "index.md",
  "module-1.md",
  "module-2.md",
  "modules.md",
]
`;

exports[`Urls should gets Urls for multiple entry points: outputFileStrategy: members 1`] = `
[
  "-internal-/index.md",
  "42-module-name/index.md",
  "basic/classes/ClassA.md",
  "basic/classes/ClassB.md",
  "basic/enumerations/EnumA.md",
  "basic/enumerations/EnumB.md",
  "basic/index.md",
  "basic/interfaces/InterfaceA.md",
  "basic/interfaces/InterfaceB.md",
  "basic/type-aliases/TypeA.md",
  "basic/type-aliases/TypeB.md",
  "has-categories/enumerations/CategoryAEnum1.md",
  "has-categories/enumerations/CategoryAEnum2.md",
  "has-categories/enumerations/CategoryBEnum1.md",
  "has-categories/enumerations/CategoryBEnum2.md",
  "has-categories/enumerations/UnCategorizedEnum.md",
  "has-categories/index.md",
  "has-categories/interfaces/CategoryAInterface1.md",
  "has-categories/interfaces/CategoryAInterface2.md",
  "has-categories/interfaces/CategoryBInterface1.md",
  "has-categories/interfaces/CategoryBInterface2.md",
  "has-categories/interfaces/UnCategorizedInterace.md",
  "has-namespaces/classes/ClassB.md",
  "has-namespaces/classes/Class_A_.md",
  "has-namespaces/enumerations/EnumB.md",
  "has-namespaces/enumerations/Enum_A_.md",
  "has-namespaces/functions/functionB.md",
  "has-namespaces/functions/function_A_.md",
  "has-namespaces/index.md",
  "has-namespaces/interfaces/InterfaceB.md",
  "has-namespaces/interfaces/Interface_A_.md",
  "has-namespaces/namespaces/NamespaceB/index.md",
  "has-namespaces/namespaces/Namespace_A_/index.md",
  "has-namespaces/namespaces/Namespace_A_/interfaces/NamespaceInterface.md",
  "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/index.md",
  "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/interfaces/NamespaceInterface.md",
  "has-namespaces/type-aliases/TypeB.md",
  "has-namespaces/type-aliases/Type_A_.md",
  "has-namespaces/variables/variableB.md",
  "has-namespaces/variables/variable_A_.md",
  "has-references/functions/defaultFunction.md",
  "has-references/index.md",
  "index/classes/SomeClass.md",
  "index/index.md",
  "index.md",
  "modules.md",
]
`;

exports[`Urls should gets Urls for multiple entry points: outputFileStrategy: members 2`] = `
[
  "-internal-/index.md",
  "42-module-name/index.md",
  "basic/classes/ClassA.md",
  "basic/classes/ClassB.md",
  "basic/enumerations/EnumA.md",
  "basic/enumerations/EnumB.md",
  "basic/index.md",
  "basic/interfaces/InterfaceA.md",
  "basic/interfaces/InterfaceB.md",
  "has-categories/enumerations/CategoryAEnum1.md",
  "has-categories/enumerations/CategoryAEnum2.md",
  "has-categories/enumerations/CategoryBEnum1.md",
  "has-categories/enumerations/CategoryBEnum2.md",
  "has-categories/enumerations/UnCategorizedEnum.md",
  "has-categories/index.md",
  "has-categories/interfaces/CategoryAInterface1.md",
  "has-categories/interfaces/CategoryAInterface2.md",
  "has-categories/interfaces/CategoryBInterface1.md",
  "has-categories/interfaces/CategoryBInterface2.md",
  "has-categories/interfaces/UnCategorizedInterace.md",
  "has-namespaces/classes/ClassB.md",
  "has-namespaces/classes/Class_A_.md",
  "has-namespaces/enumerations/EnumB.md",
  "has-namespaces/enumerations/Enum_A_.md",
  "has-namespaces/index.md",
  "has-namespaces/interfaces/InterfaceB.md",
  "has-namespaces/interfaces/Interface_A_.md",
  "has-namespaces/namespaces/NamespaceB/index.md",
  "has-namespaces/namespaces/Namespace_A_/index.md",
  "has-namespaces/namespaces/Namespace_A_/interfaces/NamespaceInterface.md",
  "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/index.md",
  "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace/interfaces/NamespaceInterface.md",
  "has-references/index.md",
  "index/classes/SomeClass.md",
  "index/index.md",
  "index.md",
]
`;

exports[`Urls should gets Urls for multiple entry points: outputFileStrategy: modules 1`] = `
[
  "-internal-.md",
  "42-module-name.md",
  "basic.md",
  "has-categories.md",
  "has-namespaces/index.md",
  "has-namespaces/namespaces/NamespaceB.md",
  "has-namespaces/namespaces/Namespace_A_/index.md",
  "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace.md",
  "has-references.md",
  "index.md",
  "module.index.md",
  "modules.md",
]
`;

exports[`Urls should gets Urls for multiple entry points: outputFileStrategy: modules 2`] = `
[
  "-internal-.md",
  "42-module-name.md",
  "basic.md",
  "has-categories.md",
  "has-namespaces/index.md",
  "has-namespaces/namespaces/NamespaceB.md",
  "has-namespaces/namespaces/Namespace_A_/index.md",
  "has-namespaces/namespaces/Namespace_A_/namespaces/NestedNamespace.md",
  "has-references.md",
  "index.md",
  "module.index.md",
]
`;

exports[`Urls should gets Urls for packages entry points: outputFileStrategy: members 1`] = `
[
  "index.md",
  "package-1/exports.md",
  "package-1/index.md",
  "package-1/interfaces/PackageInterface.md",
  "package-2/index.md",
  "package-2/module-1/index.md",
  "package-2/module-1/interfaces/PackageInterface1.md",
  "package-2/module-2/index.md",
  "package-2/module-2/interfaces/PackageInterface2.md",
  "package-3/index.md",
  "package-4/index.md",
  "package-4/interfaces/PackageInterface.md",
  "packages.md",
]
`;

exports[`Urls should gets Urls for packages entry points: outputFileStrategy: members 2`] = `
[
  "index.md",
  "package-1/index.md",
  "package-1/interfaces/PackageInterface.md",
  "package-2/index.md",
  "package-2/module-1/index.md",
  "package-2/module-1/interfaces/PackageInterface1.md",
  "package-2/module-2/index.md",
  "package-2/module-2/interfaces/PackageInterface2.md",
  "package-3/index.md",
  "package-4/index.md",
  "package-4/interfaces/PackageInterface.md",
]
`;

exports[`Urls should gets Urls for packages entry points: outputFileStrategy: modules 1`] = `
[
  "index.md",
  "package-1/exports.md",
  "package-1/index.md",
  "package-2/index.md",
  "package-2/module-1.md",
  "package-2/module-2.md",
  "package-3/index.md",
  "package-4/index.md",
  "package-4/interfaces/PackageInterface.md",
  "packages.md",
]
`;

exports[`Urls should gets Urls for packages entry points: outputFileStrategy: modules 2`] = `
[
  "index.md",
  "package-1/index.md",
  "package-2/index.md",
  "package-2/module-1.md",
  "package-2/module-2.md",
  "package-3/index.md",
  "package-4/index.md",
  "package-4/interfaces/PackageInterface.md",
]
`;

exports[`Urls should gets Urls for readme options: outputFileStrategy: members 1`] = `
[
  "functions/someFunction.md",
  "index.md",
]
`;

exports[`Urls should gets Urls for readme options: outputFileStrategy: modules 1`] = `
[
  "index.md",
]
`;

exports[`Urls should gets Urls for single entry points: outputFileStrategy: members 1`] = `
[
  "classes/AbstractClass.md",
  "classes/CallbacksOptions.md",
  "classes/ClassWithAccessors.md",
  "classes/ClassWithConstructorOverloads.md",
  "classes/ClassWithModifiers.md",
  "classes/ClassWithTypeParameters.md",
  "classes/DerivedClassA.md",
  "classes/DerivedClassB.md",
  "classes/DisposableClass.md",
  "enumerations/BasicEnum.md",
  "enumerations/EnumWithValues.md",
  "exports.md",
  "functions/basicFunction.md",
  "functions/functionReturningAFunction.md",
  "functions/functionReturningAPromise.md",
  "functions/functionReturningAnObject.md",
  "functions/functionWithDefaultParameters.md",
  "functions/functionWithMultipleSignatures.md",
  "functions/functionWithNamedParams.md",
  "functions/functionWithNestedParameters.md",
  "functions/functionWithOptionalParameters.md",
  "functions/functionWithTypeParameters.md",
  "index.md",
  "interfaces/BasicInterface.md",
  "interfaces/ExtendedInterface.md",
  "interfaces/IndexableInterface.md",
  "interfaces/InterfaceWithTypeParameters.md",
  "type-aliases/ArrayOfStuff.md",
  "type-aliases/ArrayType.md",
  "type-aliases/ConditionalType.md",
  "type-aliases/ExternalReferenceType.md",
  "type-aliases/FunctionType.md",
  "type-aliases/IndexAccessType.md",
  "type-aliases/IntersectionType.md",
  "type-aliases/LiteralType.md",
  "type-aliases/PartialMappedType.md",
  "type-aliases/PrimitiveType.md",
  "type-aliases/PromiseTypeWithObject.md",
  "type-aliases/PromiseTypeWithSymbol.md",
  "type-aliases/ReadonlyMappedType.md",
  "type-aliases/StringLiteralType.md",
  "type-aliases/TupleType.md",
  "type-aliases/TypeWithExternalSymbolLinkMapping.md",
  "type-aliases/TypeWithTypeParams.md",
  "type-aliases/UnionType.md",
  "variables/objectLiteralVariable.md",
  "variables/objectWithSymbol.md",
  "variables/stringVariable.md",
  "variables/typeOperatorVariable.md",
]
`;

exports[`Urls should gets Urls for single entry points: outputFileStrategy: members 2`] = `
[
  "classes/AbstractClass.md",
  "classes/CallbacksOptions.md",
  "classes/ClassWithAccessors.md",
  "classes/ClassWithConstructorOverloads.md",
  "classes/ClassWithModifiers.md",
  "classes/ClassWithTypeParameters.md",
  "classes/DerivedClassA.md",
  "classes/DerivedClassB.md",
  "classes/DisposableClass.md",
  "enumerations/BasicEnum.md",
  "enumerations/EnumWithValues.md",
  "functions/basicFunction.md",
  "functions/functionReturningAFunction.md",
  "functions/functionReturningAPromise.md",
  "functions/functionReturningAnObject.md",
  "functions/functionWithDefaultParameters.md",
  "functions/functionWithMultipleSignatures.md",
  "functions/functionWithNamedParams.md",
  "functions/functionWithNestedParameters.md",
  "functions/functionWithOptionalParameters.md",
  "functions/functionWithTypeParameters.md",
  "index.md",
  "interfaces/BasicInterface.md",
  "interfaces/ExtendedInterface.md",
  "interfaces/IndexableInterface.md",
  "interfaces/InterfaceWithTypeParameters.md",
  "type-aliases/ArrayOfStuff.md",
  "type-aliases/ArrayType.md",
  "type-aliases/ConditionalType.md",
  "type-aliases/ExternalReferenceType.md",
  "type-aliases/FunctionType.md",
  "type-aliases/IndexAccessType.md",
  "type-aliases/IntersectionType.md",
  "type-aliases/LiteralType.md",
  "type-aliases/PartialMappedType.md",
  "type-aliases/PrimitiveType.md",
  "type-aliases/PromiseTypeWithObject.md",
  "type-aliases/PromiseTypeWithSymbol.md",
  "type-aliases/ReadonlyMappedType.md",
  "type-aliases/StringLiteralType.md",
  "type-aliases/TupleType.md",
  "type-aliases/TypeWithExternalSymbolLinkMapping.md",
  "type-aliases/TypeWithTypeParams.md",
  "type-aliases/UnionType.md",
  "variables/objectLiteralVariable.md",
  "variables/objectWithSymbol.md",
  "variables/stringVariable.md",
  "variables/typeOperatorVariable.md",
]
`;

exports[`Urls should gets Urls for single entry points: outputFileStrategy: modules 1`] = `
[
  "exports.md",
  "index.md",
]
`;

exports[`Urls should gets Urls for single entry points: outputFileStrategy: modules 2`] = `
[
  "index.md",
]
`;
